app/logic/minor.py

```diff
@@ -0,0 +1,90 @@
+from app.models.user import User
+from app.models.course import Course
+from app.models.courseParticipant import CourseParticipant
+from app.models.program import Program
+from app.models.event import Event
+from app.models.courseInstructor import CourseInstructor
+from app.models.eventParticipant import EventParticipant
+from collections import defaultdict
+from playhouse.shortcuts import model_to_dict
+from peewee import JOIN
+
+def updateMinorInterest(username):
+    """
+    Given a username, update their minor interest and minor status.
+    """
+    user = User.get(username=username)
+    user.minorInterest = not user.minorInterest
+    if user.minorInterest == True:
+        user.minorStatus = "Interested"
+    else:
+        user.minorStatus = "No interest"
+
+    user.save()
+
+def getCourseInformation(id):
+    """
+        Given a course ID, return an object containing the course information and 
+        its instructors full names.
+    """
+
+    # retrieve the course and the course instructors
+    course = model_to_dict(Course.get_by_id(id))
+
+    courseInstructors = (CourseInstructor.select(CourseInstructor.user)
+                         .join(Course)
+                         .where(Course.id == id))
+    
+    courseInformation = {"instructors": [(instructor.user.firstName + " " + instructor.user.lastName) for instructor in courseInstructors], "course": course}
```

Comment:
This will be doing a separate query for each instructor. Join User in the query.